src/models/lineChart.js

```diff
@@ -184,7 +184,19 @@ nv.models.lineChart = function() {
                     margin.top = legend.height();
                     availableHeight1 = nv.utils.availableHeight(height, container, margin) - (focusEnable ? focusHeight : 0);
                 }
-
+                
+                //Do not display the legend if there isn't enough space
+                if (availableHeight1 === 0) {
+                    g.select('.nv-legendWrap')
+                        .datum([])
+                        .call(legend);
+
+                    if (margin.top != legend.height()) {
+                        margin.top = legend.height();
+                        availableHeight1 = nv.utils.availableHeight(height, container, margin) - (focusEnable ? focusHeight : 0);
+                    }
+                }
```

Comment:
@liquidpele ,There is not really an easy way to do that because it need many arguments but I guess we can "duplicate" this code for other charts.